routing/features_road_graph.cpp

```diff
@@ -272,6 +272,11 @@ IRoadGraph::RoadInfo const & FeaturesRoadGraph::GetCachedRoadInfo(FeatureID cons
   ri.m_speedKMPH = GetSpeedKMPHFromFt(ft);
   ft.SwapPoints(ri.m_points);
 
+  ft.ParseAltitude();
```

Comment:
So, here a DDVector reader is created for each feature. Why not to create it once for the whole mwm, not for each feature in the mwm?